### PR TITLE
Update DG: eshow implementation

### PR DESCRIPTION
**Pull request by ruo-x**
added diagrams for some steps but not sure if they are good enough thoughts? 
image
image
image
image

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -310,20 +310,70 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+Step 3. The user executes `eshow 1` to read more about the first event created.
+
+The `eshow` command obtains the event at index `1` from the last shown list (`filteredEventList`) as the index is based on the most recent event list displayed to the user before he calls the `eshow` command. This event is the one that will be shown to the user in the pop-up window.
 
-When the app is initialized, the MainApp will first initialize all key components of the app. Specifically in this case:
+The `eshow` command then filters the `eventToShowList` such that it has only one element, which is the event previously obtained.
 
-First, eventToShowList in ModelManager will be initialized, with the list of all events currently in the EventStorage.
+The `eshow` command then returns a `CommandResult` object with its `isShowEvent` field as true.
```
Comment (zacwong2151) [inline docs/DeveloperGuide.md:333]: hmm I would say be careful of putting in low-level details in your DG, as readers prolly want to see the high-level details

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -310,20 +310,70 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+### Reading an individual event feature
 
-#### Proposed Implementation
+#### Implementation
+
+When event show command is called, a pop-up window (an `EventShowWindow` object) will show, displaying the information of the event that corresponds to the given index. The event to show is obtained from the `eventToShowList` instantiated in `ModelManager`.
+
+Given below is an example usage scenario and how the event show mechanism behaves at each step:
+
+Step 1. The user launches the app for the first time, and executes `ecreate n/first event ...` to create a new event. This called method `Model#updateFilteredEventList()` to update both the `filteredEventList` and the `eventToShowList`.
```
Comment (zacwong2151) [inline docs/DeveloperGuide.md:321]: `This called method` --> `This calls the method`

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -310,20 +310,70 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+The `eshow` command obtains the event at index `1` from the last shown list (`filteredEventList`) as the index is based on the most recent event list displayed to the user before he calls the `eshow` command. This event is the one that will be shown to the user in the pop-up window.
 
-When the app is initialized, the MainApp will first initialize all key components of the app. Specifically in this case:
+The `eshow` command then filters the `eventToShowList` such that it has only one element, which is the event previously obtained.
 
-First, eventToShowList in ModelManager will be initialized, with the list of all events currently in the EventStorage.
+The `eshow` command then returns a `CommandResult` object with its `isShowEvent` field as true.
 
-Then, when MainWindow is being initialized, the EventShowWindow will also be initialized.
+The `MainWindow` received the `CommandResult`, identifies that its `isShowEvent` field is true, hence executing the `MainWindow#handleShowEvent()` method, which shows/focuses the `EventShowWindow`, after loading its contents by calling `EventShowWindow#loadContents()`.
```
Comment (zacwong2151) [inline docs/DeveloperGuide.md:335]: e.g. here might be too many low-level details, others might wanna chime in too

**Review by zacwong2151 — COMMENTED** [review]
some concerns

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (ruo-x) [reply docs/DeveloperGuide.md:333]: oki will go and look at it again xiele

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (ruo-x) [reply docs/DeveloperGuide.md:321]: yes english dashen

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (ruo-x) [reply docs/DeveloperGuide.md:341]: hmm i gets ya but im not sure what to include and what not to as it kinda explains the mechanism behind eshow, any suggestions?

**Review by jordankanghm — CHANGES_REQUESTED** [review]
bad job

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -310,20 +310,83 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+Step 2. The user  executes `ecreate n/second event ...` to create another event. This called the method `Model#updateFilteredEventList()` again to update both the `filteredEventlist` and the `eventToShowList`.
+
+<puml src="diagrams/EventShowStep2.puml" alt="EventShowStep2" />
+
+Displayed event list now has two events at index 1 and 2 respectively.
+
+Step 3. The user executes `eshow 1` to read more about the first event created.
+
+The `eshow` command obtains the event at index `1` from the last shown list (`filteredEventList`) as the index is based on the most recent event list displayed to the user before he calls the `eshow` command. This event is the one that will be shown to the user in the pop-up window.
```
[the remark this answers is not in the document]
Comment (tiongMax) [reply docs/DeveloperGuide.md:333]: ahhahah u cb

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (zacwong2151) [reply docs/DeveloperGuide.md:339]: actl I realised what i said not really true LOL ur level of details is fine

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (zacwong2151) [reply docs/DeveloperGuide.md:341]: This sentence at line 341 has too many commas, shd split it up into a few sentences to make it more understandable for the reader

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -310,20 +310,83 @@ The following activity diagram summarizes what happens when a user executes a ne
...
 
+User see a pop-up window (`EventShowWindow`), showing all information of the event at index `1`.
+
+Step 4. The user navigates back to the main window, causing the pop-up window to be unfocused.
+
+Step 5. The user now wants to read more on the second event, so he executes `eshow 2`. 
+
+Step 3 is repeated with the new given index - `2`.
+
+<puml src="diagrams/EventShowStep5.puml" alt="EventShowStep5" />
```
Comment (zacwong2151) [inline docs/DeveloperGuide.md:355]: u have a step 5 but no step 4

**Review by zacwong2151 — APPROVED** [review]
one lil thing, other than that ur diagrams look mega g

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (ruo-x) [reply docs/DeveloperGuide.md:339]: rly ah ok good

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (ruo-x) [reply docs/DeveloperGuide.md:341]: k english teacher

Comment (ruo-x) [reply docs/DeveloperGuide.md:341]: @jordankanghm is the diagram i screenshot and put on top sufficient?

File: docs/DeveloperGuide.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (ruo-x) [reply docs/DeveloperGuide.md:333]: i want he cannot ah i like men

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (ruo-x) [reply docs/DeveloperGuide.md:355]: have la

**Review by jordankanghm — APPROVED** [review]
wah solid

**Pull request merged**